django/apps/aggregated/management/commands/update_aggregated_data.py

```diff
@@ -22,44 +22,44 @@
         swipes
     )
     (
-        -- Retrieve used tasks
-        WITH used_tasks as (
+        -- Retrieve used task groups
+        WITH used_task_groups as (
             SELECT
-              project_id, group_id, task_id
-            FROM results R
-              INNER JOIN tasks T USING (project_id, group_id, task_id)
+              MS.project_id, MS.group_id
+            FROM mapping_sessions MS
             WHERE
-                R.timestamp >= %(from_date)s and R.timestamp < %(until_date)s
-            GROUP BY project_id, group_id, task_id
+                MS.start_time >= %(from_date)s and MS.start_time < %(until_date)s
+            GROUP BY project_id, group_id -- To get unique
         ),
-        -- Calculated task area.
-        task_data as (
+        -- Calculated area by task_groups
+        task_group_data as (
           SELECT
               project_id,
               group_id,
-              task_id,
-              ST_Area(geom::geography(GEOMETRY,4326)) / 1000000 as area -- sqkm
-          FROM used_tasks
-              INNER JOIN tasks T USING (project_id, group_id, task_id)
+              SUM(
+                ST_Area(geom::geography(GEOMETRY,4326)) / 1000000
```

Comment:
4326 is from the pre-existing table definition. Let me know if I need to add this. (The URL is long, want to avoid breaking it into lines, with just branch it might change in future)

https://github.com/mapswipe/python-mapswipe-workers/blob/c80cc6eef6f99bc762096b532857c92205421e77/postgres/initdb.sql#L47

Reply:
@Hagellach37 not a blocking point, but your input is probably needed here. is 4326 the projection type? Maybe worth documenting somewhere?

Reply:
yes 4326  refers to the projection, e.g. check here: https://epsg.io/4326 and is the one using latitude, longitude in degrees.

It's the most common projection used and also the one we use to store our geometries. the important part of that query is that we cast the geometry as  `geography` type which will ensure that we get the area in square meters. Otherwise you would end up with the wrong values for area as it would be in degrees which would make no sense.